umbraco/Umbraco-CMS
# [v14] Fix multiple upgrader issues

### Prerequisites


### Fixed issues
- [x] An error is thrown during unattended upgrade because of changes to the preview hub that could no longer deal with full cache refreshes (no key is present)
- [x] When the migrations changes the user in such a way that the link between the already issued tokens and the user store object is broken, authorization will throw server errors => a revoke token mechanism is introduced that can be called by the upgrader when a migration deems it necessary.

### Open related issues
- After login to authorize the upgrade, the user is redirected to the back office instead of the upgrade screen. This results in the back office being in a bad state due to migrations not having ran yet. This can be circumvented by refreshing the page.

### Testing
- create a new v13 latest database with 1 custom datatype/contenttype/document
- use this database to run the upgrader both unattended and attended. The only issue that should remain is having to refresh the page after login as described above.
